apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -702,15 +705,16 @@ handle_ch_status_changed(ChId, #{ last_block := Last
     lager:debug("channel status changed, Hash=~p, Last=~p", [Hash,Last]),
     {TxLog, C1} = get_txs_since({any_after_block, Last}, Hash, ChId, C),
     lager:debug("New TxLog: ~p", [TxLog]),
-    #{tx_hash := Tx, block_hash := BlockHash} = get_latest_tx(TxLog),
-    lager:debug("Latest Tx = ~p, Hash = ~p", [Tx, BlockHash]),
-    {Status, C2} = get_basic_ch_status_(ChId, BlockHash, C1),
+    {TxHash, #{tx := Tx, block_hash := BlockHash}, C2} =
```

Comment:
I need to understand how this optional `tx` field is guaranteed to be present here 🤔

Reply:
See `ensure_signed_tx_included/2`, called from line 793.